synthesizer/process/src/stack/mod.rs

```diff
@@ -203,10 +210,12 @@ pub struct Stack<N: Network> {
     program: Program<N>,
     /// A reference to the global stack map.
     stacks: Weak<RwLock<IndexMap<ProgramID<N>, Arc<Stack<N>>>>>,
+    /// The register types for the program constructor, if it exists.
+    constructor_types: Arc<RwLock<Option<FinalizeTypes<N>>>>,
```

Comment:
Can you explain why locks are needed?

Reply:
I specifically went with `RwLock` instead of `RefCell` to be consistent with the proving and verifying keys.